spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/WebOperationHttpService.java

```diff
@@ -0,0 +1,133 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+/**
+ * {@link HttpService} to handle a {@link WebOperation}. Mostly inspired by reactive implementation in
+ * {@link org.springframework.boot.actuate.endpoint.web.reactive.AbstractWebFluxEndpointHandlerMapping}.
+ */
+final class WebOperationHttpService implements HttpService {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<Map<String, Object>> JSON_MAP =
+            new TypeReference<Map<String, Object>>() {};
+
+    private final WebOperation operation;
+
+    WebOperationHttpService(WebOperation operation) {
+        this.operation = operation;
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) {
+        final CompletableFuture<HttpResponse> resFuture = new CompletableFuture<>();
+        req.aggregate().handle((msg, t) -> {
+            if (t != null) {
+                resFuture.completeExceptionally(t);
+                return null;
+            }
+            if (operation.isBlocking()) {
+                ctx.blockingTaskExecutor().execute(() -> invoke(ctx, msg, resFuture));
+            } else {
+                invoke(ctx, msg, resFuture);
+            }
+            return null;
+        });
+        return HttpResponse.from(resFuture);
+    }
+
+    private void invoke(ServiceRequestContext ctx,
+                        AggregatedHttpMessage msg,
+                        CompletableFuture<HttpResponse> resFuture) {
+        final Map<String, Object> arguments = getArguments(ctx, msg);
+        final Object result = operation.invoke(new InvocationContext(SecurityContext.NONE, arguments));
+
+        try {
+            final HttpResponse res = handleResult(result, msg.method());
+            resFuture.complete(res);
+        } catch (IOException e) {
+            resFuture.completeExceptionally(e);
+        }
+    }
+
+    private static HttpResponse handleResult(@Nullable Object result, HttpMethod method) throws IOException {
+        if (result == null) {
+            return HttpResponse.of(method != HttpMethod.GET ? HttpStatus.NO_CONTENT : HttpStatus.NOT_FOUND);
+        }
+
+        if (!(result instanceof WebEndpointResponse)) {
+            return HttpResponse.of(HttpStatus.OK,
+                                   MediaType.JSON_UTF_8,
+                                   OBJECT_MAPPER.writeValueAsBytes(result));
```

Comment:
eeek.. this quotes prometheus scrape data, making it invalid

org.springframework.boot.actuate.metrics.export.prometheus.PrometheusScrapeEndpoint produces text/plain

```
  @ReadOperation(
    produces = {"text/plain; version=0.0.4; charset=utf-8"}
  )
```

Reply:
the response of the server returns the wrong media type which would be correct if it was json..

```bash
$ curl -vs http://localhost:9411/actuator/prometheus|head -c 140
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9411 (#0)
> GET /actuator/prometheus HTTP/1.1
> Host: localhost:9411
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< content-type: application/json; charset=utf-8
< content-length: 14017
< 
{ [14017 bytes data]
* Connection #0 to host localhost left intact
"# HELP jvm_threads_daemon_threads The current number of live daemon threads\n# TYPE jvm_threads_daemon_threads gauge\njvm_threads_daemon_th
```